ui/src/main/java/edu/wpi/grip/ui/util/deployment/HeadlessInstanceManager.java

```diff
@@ -0,0 +1,155 @@
+package edu.wpi.grip.ui.util.deployment;
+
+
+import com.google.common.eventbus.EventBus;
+import com.jcabi.ssh.Shell;
+import com.jcraft.jsch.JSch;
+import edu.wpi.grip.core.StartStoppable;
+import edu.wpi.grip.core.events.StartedStoppedEvent;
+import edu.wpi.grip.core.events.UnexpectedThrowableEvent;
+import org.apache.tools.ant.BuildException;
+import org.apache.tools.ant.Project;
+import org.apache.tools.ant.taskdefs.optional.ssh.Scp;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URLDecoder;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Controls an instance of GRIP running on a remote device.
+ */
+public class HeadlessInstanceManager implements StartStoppable {
+
+    private final EventBus eventBus;
+    private final File coreJar;
+    private final File projectFile;
+    private final SecureShellDetails details;
+    private final DeploymentCommands deploymentCommands;
+    private Optional<Thread> sshThread;
+
+    /**
+     *
+     * @param eventBus
+     * @param coreJar The jar with all of the core. This will be copied to the destination
+     * @param projectFile The project file to send over
+     * @param details The details regarding connecting to the secure shell
+     * @param deploymentCommands The commands required to start and stop GRIP
+     */
+    public HeadlessInstanceManager(EventBus eventBus, File coreJar, File projectFile, SecureShellDetails details, DeploymentCommands deploymentCommands) {
```

Comment:
This class should be hardcoded to only do FRC.  Abstracting away certain parts of it will not make it any easier to port it to other platforms, since FRC is the only one where we have an exact set of commands that we know will work.

Of course, more functionality can be added later, but for now this is [premature encapsulation](http://www.adam-bien.com/roller/abien/entry/premature_encapsulation_is_the_root).

9/10